mocks/mockstub/mock_stub.go

```diff
@@ -0,0 +1,188 @@
+package mockstub
+
+import (
+	"encoding/hex"
+	"testing"
+	"time"
+
+	"github.com/anoideaopen/foundation/core"
+	"github.com/anoideaopen/foundation/mocks"
+	pbfound "github.com/anoideaopen/foundation/proto"
+	"github.com/btcsuite/btcd/btcutil/base58"
+	"github.com/golang/protobuf/proto" //nolint: staticcheck
+	"github.com/google/uuid"
+	"github.com/hyperledger/fabric-chaincode-go/shim"
+	"github.com/hyperledger/fabric-protos-go/peer"
+	"github.com/stretchr/testify/require"
+)
+
+type MockStub struct {
+	stub             *mocks.ChaincodeStub
+	getStateCallsMap map[string][]byte
+}
+
+// NewMockStub returns new mock stub
+func NewMockStub(t *testing.T) *MockStub {
+	mockStub := new(mocks.ChaincodeStub)
+	txID := [16]byte(uuid.New())
+	mockStub.GetTxIDReturns(hex.EncodeToString(txID[:]))
+	mockStub.GetSignedProposalReturns(&peer.SignedProposal{}, nil)
+
+	err := mocks.SetCreatorCert(mockStub, mocks.TestCreatorMSP, mocks.AdminCert)
+	require.NoError(t, err)
+
+	mockStub.CreateCompositeKeyCalls(shim.CreateCompositeKey)
+	mockStub.SplitCompositeKeyCalls(func(s string) (string, []string, error) {
+		componentIndex := 1
+		var components []string
+		for i := 1; i < len(s); i++ {
+			if s[i] == 0 {
+				components = append(components, s[componentIndex:i])
+				componentIndex = i + 1
+			}
+		}
+		return components[0], components[1:], nil
+	})
+
+	getStateCallsMap := make(map[string][]byte)
+
+	mockStub.GetStateCalls(func(key string) ([]byte, error) {
+		value, ok := getStateCallsMap[key]
+		if ok {
+			return value, nil
+		}
+
+		return nil, nil
+	})
+
+	mockStub.InvokeChaincodeCalls(func(chaincodeName string, args [][]byte, channelName string) peer.Response {
+		if chaincodeName != "acl" && channelName != "acl" {
+			return shim.Error("mock stub does not support chaincode " + chaincodeName + " and channel " + channelName + " calls")
+		}
+		functionName := string(args[0])
+
+		parameters := make([]string, 0, len(args[1:]))
+		for _, arg := range args[1:] {
+			parameters = append(parameters, string(arg))
+		}
+
+		switch functionName {
+		case FnCheckAddress:
+			return MockACLCheckAddress(parameters[0])
+		case FnCheckKeys:
+			return MockACLCheckKeys(parameters[0])
+		case FnGetAccountInfo:
+			return MockACLGetAccountInfo()
+		case FnGetAccountsInfo:
+			return MockACLGetAccountsInfo()
+		}
```

Comment:
This code should also be translated into map so that you can add your own handlers